aries_vcx/src/new_protocols/connection/inviter/mod.rs

```diff
@@ -0,0 +1,25 @@
+pub mod handlers;
+pub mod state;
+
+use diddoc_legacy::aries::diddoc::AriesDidDoc;
+
+use self::state::InviterComplete;
+
+// If this flies with a single state then
+// we can remove the generic
+#[derive(Clone, Debug)]
+pub struct InviterConnection<S> {
+    did: String,
+    verkey: String,
+    state: S,
+}
+
+impl InviterConnection<InviterComplete> {
```

Comment:
### ELI5 observatory comment for other reviewers:

Bogdan's approach reduced number of state significantly - one reason for that is elimination of states such as `Invited` stated, which used to be a state where a "invitation message has been received and somewhat validated", and the state held the invitation.

This was somewhat convenient for the consumers, because upon receiving invitation, they would construct "connection state machine", 1 type of object they would deal the entire time, anytime they need to do something / drive protocol further. You could have a table or a collection of Connection objects, track their state and try to drive them forward asynchronously over time, if you wanted.

With Bogdan's approach such "message holder" state are gone - if you receive invitation, and you want to drive the protocol later on (for example after mobile app restart, server restart, etc.) it's your responsibility to have a place for message store. 
For example, you will need to store Invitation, and then pick it up later on if you want to proceed. It's simple with connection, but can get more complex when you are already in the middle of protocol. For example you are trying to receive a credential. You have sent `credential-request` message, so you already have a state machine in some "Requested" state. Next up you receive `issuer-credential`, but you only want to proceed with human supervision. So if you want to enable that being asynchronous, or span multiple servers, you need to store the `issuer-credential` message along with the `requested` state machine. You either link up the 2 objects, or simply anytime you pick-up unprocessed message from your store, you resolve the appropriate state machine (probably a lot better approach).

### Opinion
Although some places we need to imo keep some granularity in number of state, this approach forces consumers to work bit more with message, but actually this plugs in kind of nicely in mobile setting with mediators, which literally provide you the "store of unprocessed messages" where many message types trigger some UI user prompt. After the transition is completed, typically the message would be marked in the mediator as processed / marked for deletion.

In server setting there might likely be some queue middleware delivering the messages the processors and it's kind of the same story.

The only thing you need it store of **unprocessed** messages, and from a message you then always resolve relevant state machine, do some processing, trigger a prompt, after transition is done, delete the message (or keep it for some message audit-log purposes)